RELEASE_PROCESS.md

```diff
@@ -20,92 +40,278 @@ Regular releases concern only major and minor releases. We do not decide up fron
 
 Patch releases are automatically released and published.
 
-## Release phases
+### Release date philosophy
+
+The release is scheduled for a given month, not a specific day. We will release once everything is ready, and won't wait until the last day of the month with the release.
+
+We release often, so anything that isn't ready can wait until the next release (e.g. it can be released in June if the change will not make it in April). Our priority is the quality, and not quantity, of the features. Quality means that changes for a release is properly delivered in as many tools as possible with very good documentation and examples.
+
+---
+## What?
+
+### Repositories
+
+Github repositories that need to be updated at the same time to enable a release, in a coordinated way, are:
+- [spec](https://github.com/asyncapi/spec) - the specification file
+- [spec-json-schemas](https://github.com/asyncapi/spec-json-schemas) - the JSON schema
+- [parser-js](https://github.com/asyncapi/parser-js) - the JavaScript parser
+- [website](https://github.com/asyncapi/website) - the website
+
+---
+## How?
+
+The steps described here are not intended to be strictly chronological. For a single release, it might make sense to start some pieces of work in a slightly different order. The intention of the steps here is meant to give release coordinators a rough idea of the sequence involved.
+
+### Step 1 - kick off call
+
+A nice way to start a new release is to do one or more of the initial steps on an open live call/meeting.
+
+This isn't a required step, but possible benefits include:
+- raising awareness of the upcoming release
+- giving a new release coordinator support and initial direction to help them get started
+- allowing potential future release coordinators to see what is involved
+
+Examples of arranging a contribution call like this can be found in the [community repository](https://github.com/asyncapi/community/issues?q=is%3Aissue+label%3Ameeting+contributor-first+)
+
+
+### Step 2 - create a release issue
+
+To allow the community to see what is currently being considered for inclusion in a release, and to give easy, at-a-glance visibility over the progress, create a new Github issue in the [spec](https://github.com/asyncapi/spec) repository.
+
+This issue should be updated as each of the following steps are completed.
+
+A [template for a new release issue](https://github.com/asyncapi/spec/blob/master/.github/ISSUE_TEMPLATE/release.md) is available.
+
+An example is the [release issue for the 2.3.0](https://github.com/asyncapi/spec/issues/675) release.
 
-### Kick off
+### Step 3 - create release branches
 
-We start by creating release branches and a placeholder for release notes in the AsyncAPI Blog.
+Release branches must have a year and a month of the release as prefix: {YEAR_OF_RELEASE}-{MONTH_OF_RELEASE}-release.
```

Comment:
I suggest we follow semantic release `next` and `next-major` branch names.

I suggested it in the past in Slack as well: https://asyncapi.slack.com/archives/C0230UAM6R3/p1644418043568169

Reply:
I ended up creating an issue for my suggestion: https://github.com/asyncapi/spec/issues/734